cmake/IRODSConfigVersion.cmake.in

```diff
@@ -0,0 +1,158 @@
+# for development builds. For example, 4.97 would be a development version between 4.3 and 5.0,
+# and 5.92 would be a development version between 5 and 6.0.
+# Starting with iRODS 5.0, this extends to minor releases as well. For example, 5.1.94 would be a
+# development version between 5.1 and 5.2.0.
+#
+# Thusly, our CMake package version file is designed as so:
+# - An iRODS installation should match as exactly compatible if the find_package()-calling project
+#   is searching for the exact same version.
+# - An iRODS installation should match as compatible if the find_package()-calling project is
+#   searching for a version range containing the version of the iRODS installation.
+# - An iRODS installation without a .9x version should match as potentially compatible if the
+#   find_package()-calling project is searching for an earlier version of the same major release
+#   or a .9x version of the previous major release.
+# - An iRODS installation with a .9x version in the second part of the version number should match
+#   as potentially compatible if the find_package()-calling project is searching for the next
+#   major release or any earlier version of the same major release.
+# - An iRODS installation with a .9x version in the third part of the version number should match
+#   as potentially compatible if the find_package()-calling project is searching for the next
+#   minor release or any earlier version of the same major release.
+
+set(IRODS_VERSION_1 "@IRODS_VERSION_MAJOR@")
+set(IRODS_VERSION_2 "@IRODS_VERSION_MINOR@")
+set(IRODS_VERSION_3 "@IRODS_VERSION_PATCH@")
+set(PACKAGE_VERSION "${IRODS_VERSION_1}.${IRODS_VERSION_2}.${IRODS_VERSION_3}")
+
+# Get rid of any leading zeroes. (There probably won't ever be any, but best to check anyway.)
+if(NOT IRODS_VERSION_1 VERSION_EQUAL "0")
+  string(REGEX REPLACE "^0+" "" IRODS_VERSION_1 "${IRODS_VERSION_1}")
+endif()
+if(NOT IRODS_VERSION_2 VERSION_EQUAL "0")
+  string(REGEX REPLACE "^0+" "" IRODS_VERSION_2 "${IRODS_VERSION_2}")
+endif()
+if(NOT IRODS_VERSION_3 VERSION_EQUAL "0")
+  string(REGEX REPLACE "^0+" "" IRODS_VERSION_3 "${IRODS_VERSION_3}")
+endif()
+
+# Check version range first.
+if (PACKAGE_FIND_VERSION_RANGE)
+  # The way the documentation is written implies that PACKAGE_FIND_VERSION_RANGE_MIN might someday support "EXCLUDE"
+  # and supporting that possibilty here is trivial, so let's do it.
+  if (
+    ((PACKAGE_FIND_VERSION_RANGE_MIN STREQUAL "INCLUDE" AND PACKAGE_VERSION VERSION_LESS PACKAGE_FIND_VERSION_MIN)
+      OR (PACKAGE_FIND_VERSION_RANGE_MIN STREQUAL "EXCLUDE" AND PACKAGE_VERSION VERSION_LESS_EQUAL PACKAGE_FIND_VERSION_MIN))
+    OR ((PACKAGE_FIND_VERSION_RANGE_MAX STREQUAL "INCLUDE" AND PACKAGE_VERSION VERSION_GREATER PACKAGE_FIND_VERSION_MAX)
+      OR (PACKAGE_FIND_VERSION_RANGE_MAX STREQUAL "EXCLUDE" AND PACKAGE_VERSION VERSION_GREATER_EQUAL PACKAGE_FIND_VERSION_MAX))
+  )
```

Comment:
Took me a while to understand what this was doing. Please leave a comment explaining what this does for the next person who ends up here.

Reply:
Done